### DISCUSSION TITLE
Comparing Absolute Distance using GlobalTransform Not Working as Expected

### DISCUSSION THREAD

**Discussion by TomPridham** [bevyengine/bevy#4430] [Q&A] [answered]
I'm trying to implement a rope using Verlet transforms and enforcing constraints via Jakobsen's method. I was working off this post: https://www.owlree.blog/posts/simulating-a-rope.html. Things were working mostly correctly when I was using normal `Transform`. The main issue there was that the first(parent) node and the second(child) node weren't offset correctly because the `Transform` for the parent is relative to the origin and the child is relative to the parent. I thought that using `GlobalTransform` to compare the distances would be the correct thing to do, but now the translation for the cubes I'm rendering very quickly approach the max for a f32, overflow and disappear. I could solve this by have a separate parent element that is basically just a container for the rope elements, but that doesn't seem like it should be required. Is that just the way?

You can see what I am talking about here: https://github.com/TomPridham/bevy-cable/pull/3/files. The pull request has the bad behavior using `GlobalTransform` and main has the normal behavior using `Transform`. You can run `cargo run --example basic` to see it in action. The problem of the translation being offset because of the parent isn't present in the example because the parent is centered around the origin

**Comment by DJMcNab** [comment]
I haven't done any debugging of the code, but when your cables in the child move the parent, they will also change their own `GlobalTransform` by the same amount, effectively cancelling out the move on the parent.

This seems like a possible mechanism for things to zoom off.

**Reply by TomPridham** [reply]
I don't think that's it. I just tried spawning them all without a parent and saw the same behavior

**Comment by TomPridham** [comment, marked as answer]
I think it's related to the last paragraph here: https://bevy-cheatbook.github.io/features/transforms.html. The system is fine for a single iteration, but gradually gets more and more out of sync because the `Transform`s are getting changed twice for every `GlobalTransform` update. i think i will just have a container element that isn't technically part of the cable and use normal `Transform`s to change everything
